src/Option.purs

```diff
@@ -1229,6 +1230,49 @@ set proxy value = modify proxy go
   go :: forall a. a -> value
   go _ = value
 
+-- | Changes a key with the given value (if it exists) to an option.
+-- | As with `set`, the key must already exist in the option.
+-- |
+-- | E.g.
+-- | ```PureScript
+-- | someOption :: Option.Option ( foo :: Boolean, bar :: Int )
+-- | someOption = Option.empty
+-- |
+-- | anotherOption :: Option.Option ( foo :: Boolean, bar :: Int )
+-- | anotherOption = Option.setMay (Data.Symbol.SProxy :: _ "bar") (Just 31 someOption
+-- | ```
+-- |
+-- | The `proxy` can be anything so long as its type variable has kind `Symbol`.
+-- |
+-- | It will commonly be `Data.Symbol.SProxy`, but doesn't have to be.
+setMay ::
```

Comment:
I'm down with this, but I'd like to think about a different name before making a new release. I'm pretty partial to longer names most times.

Reply:
sure!